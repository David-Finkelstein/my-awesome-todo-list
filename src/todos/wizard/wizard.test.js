import React from 'react';
import { mount } from "enzyme";
import immutable from "immutable";
import { Provider } from "react-redux";
import { Router } from "react-router-dom";
import { mountToJson } from "enzyme-to-json";
import { createBrowserHistory } from 'history';
import { createStore, combineReducers } from 'redux';

import TodoWizard from './index';

const history = createBrowserHistory();
const reducer = combineReducers({ todoList: () => { return immutable.fromJS({}) } });
const store = createStore(reducer, undefined, undefined);
const mockFn = jest.fn();
const mockProps = {
    callback: mockFn,
    title: '',
    btnText: '',
};

describe('<TodoWizard />', () => {
    it('checks the functionality of delete/edit item according to the callback function that passed', () => {
        const wrapper = mount(
            <Provider store={store}>
                <Router history={history}>
                    <TodoWizard {...mockProps}/>
                </Router>
            </Provider>
        );

        const deleteOrEditBtn = wrapper.find('Button').at(0);
        expect(mockProps.callback.mock.calls.length).toBe(0);
        deleteOrEditBtn.props().onClick();
        expect(mockProps.callback.mock.calls.length).toBe(1);

    });

    it('takes a snapshot of the component and check if the prev snapshot equals to the current', () => {
        const wrapper = mount(
            <Provider store={store}>
                <Router history={history}>
                    <TodoWizard {...mockProps}/>
                </Router>
            </Provider>);
        expect(mountToJson(wrapper.debug())).toMatchSnapshot();
    });
});