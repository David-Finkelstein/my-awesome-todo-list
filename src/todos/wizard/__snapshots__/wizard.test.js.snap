// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoWizard /> takes a snapshot of the component and check if the prev snapshot equals to the current 1`] = `null`;
